nuxeo-services/nuxeo-platform-filemanager-core-listener/src/test/java/org/nuxeo/ecm/platform/filemanager/core/listener/tests/TestMimetypeIconUpdater.java

```diff
@@ -102,80 +115,128 @@ public void testMimeTypeUpdaterWithoutPrefix() {
      */
     @Test
     public void testMimeTypeUpdaterWithPrefix() {
-        DocumentModel doc = createWithPrefixBlobDocument(false);
+        DocumentModel doc = createDocument("SimpleBlobDocument", PDF_EXTENSION, null);
         Blob blob = (Blob) doc.getProperty("simpleblob", "blob");
         assertNotNull(blob);
         String mt = blob.getMimeType();
         assertNotNull(mt);
-        assertEquals("application/pdf", mt);
+        assertEquals(PDF_MIMETYPE, mt);
     }
 
     @Test
     public void testMimeTypeUpdaterFolderish() {
         // Workspace is folderish and contains the file schema
-        DocumentModel doc = createWorkspace();
+        DocumentModel doc = createDocument("Workspace", PDF_EXTENSION, null);
         Blob blob = (Blob) doc.getProperty("file", "content");
         assertNotNull(blob);
         String mt = blob.getMimeType();
         assertNotNull(mt);
-        assertEquals("application/pdf", mt);
+        assertEquals(PDF_MIMETYPE, mt);
 
         String icon = (String) doc.getProperty("common", "icon");
         assertNull(icon); // default icon, not overridden by mime type
     }
 
-    protected DocumentModel createWithoutPrefixBlobDocument(boolean setMimeType) {
-        DocumentModel withoutPrefixBlobDoc = coreSession.createDocumentModel("/", "testFile", "WithoutPrefixDocument");
-        withoutPrefixBlobDoc.setProperty("dublincore", "title", "TestFile");
+    /**
+     * Ensure that the document blob mime type is normalised if its possible, even if the current blob have a mime type.
+     * 
+     * @since 11.1
+     */
+    @Test
+    public void shouldNormaliseMimeTypedIfItsPossible() {
+        DocumentModel documentModel = createDocument("File", PSD_EXTENSION, "image/photoshop");
```

Comment:
Not sure why the ".psd" extension is important here, since you give the blob mime type (then the extension shouldn't matter).

Reply:
yes it's not important, but i did it for more clearty. like this we see that we are going to create a blob which is a psd file. Like in real case; we have an extension and the mime type. in other hand if i didn't give the extension, an other developer can think i am testing other thing.